openrosa-form-submission.rst

```diff
@@ -0,0 +1,162 @@
+Form Submission API
+======================
+
+This standard defines the API for submitting data to an OpenRosa compliant server and receiving a response from the server about the submission. This API is designed to provide a base level of interoperability between OpenRosa clients and servers while enabling application-specific extensions as well.
+
+.. _form-submission:
+
+Form Submission
+--------------------
+
+OpenRosa servers must be able to receive XForms which are submitted by clients over HTTP. 
+
+There are 3 major categories of requirements for submission which must be fulfilled: 
+
+- channel
+- content
+- correctness.
+
+.. _channel:
+
+Channel
+~~~~~~~~~~~~
+
+OpenRosa servers MUST provide a URI capable of accepting HTTP POST Requests. Access controls, firewalls and geographic or networking restrictions MAY limit the origins and/or clients that can access the server.
+
+The server MUST support HTTP 1.1 chunked transfer encoding for receiving the POST content. There is no minimum set of standards for timeouts, maximum content length, or other http configurations, but servers are encouraged to support lenient connections and the largest possible content size, since many of the connections will be from unreliable channels and in environments where splitting content up is impractical.
+
+For maximum compatibility with J2ME clients, it is recommended that a server SHOULD NOT issue redirects. 
+
+.. note::
+  Using digest authentication and https when communicating with a server does not require any redirects --- you can have authentication and secure transport without redirects.
+
+.. _content:
+
+Content
+~~~~~~~~~~
+
+OpenRosa submissions (and responses) MUST provide headers conforming to the OpenRosa Request standard.
+
+Successful server responses MUST include an ``X-OpenRosa-Accept-Content-Length`` header in addition to the :ref:`required OpenRosa Request response headers <openrosa-responses>`. The ``X-OpenRosa-Accept-Content-Length`` header specifies this server's recommended maximum size for a POST body, in bytes.
+
+Servers SHOULD include this header in their error responses. However, clients MUST NOT rely on the presence of this header (or any OpenRosa header) in every error response.
+
+.. warning:: 
```

Comment:
I don't know if this should be a warning. It's more of an example of a case in which a header wouldn't be provided.

Reply:
changed it to `note`